### PR TITLE
Adds doc comments for relational/query.

### PR DESCRIPTION
**Pull request by anpete**
cc @rowanmiller

### CODE REVIEW

File: src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/AggregateExpression.cs [hunk 1]
```diff
@@ -7,21 +7,52 @@
...
+        ///     Throws an exception if the node isn't reducible.
+        /// </summary>
+        /// <param name="visitor"> An instance of <see cref="ExpressionVisitor" />. </param>
+        /// <returns> The expression being visited, or an expression which should replace it in the tree. </returns>
+        /// <remarks>
+        ///     Override this method to provide logic to walk the node's children.
+        ///     A typical implementation will call visitor.Visit on each of its
+        ///     children, and if any of them change, should return a new copy of
+        ///     itself with the modified children.
+        /// </remarks>
```
Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/AggregateExpression.cs]: Remarks don't show up in VS IntelliSense, so if this is info that folks would want when coding against the API I would move it to a <para> in the <summary>. Your call though, I'm fine either way.

Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/AggregateExpression.cs]: This is just copied from the base BCL docs.

Comment (rowanmiller) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/Expressions/AggregateExpression.cs]: Sounds good to leave as-is then

**Comment by rowanmiller** [comment]
:shipit: from my perspective. There is a bit of code refactoring in this, which all looked fine to me, but maybe @divega wants to take a look over that.

**Comment by divega** [comment]
Will do

File: src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionTranslators/SingleOverloadStaticMethodCallTranslator.cs [hunk 2]
```diff
@@ -10,12 +10,22 @@
 
 namespace Microsoft.EntityFrameworkCore.Query.ExpressionTranslators
 {
+    /// <summary>
+    ///     A base LINQ expression translator for CLR <see cref="MethodCallExpression" /> expressions that
+    ///     have multiple overloads.
```
Comment (divega) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionTranslators/SingleOverloadStaticMethodCallTranslator.cs]: Did you mean the opposite?

Comment (anpete) [inline src/Microsoft.EntityFrameworkCore.Relational/Query/ExpressionTranslators/SingleOverloadStaticMethodCallTranslator.cs]: Well, it is correct for the class it came from :smile:

**Comment by divega** [comment]
:shipit:

**Pull request merged**